src/context/analytics.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useContext } from "react";
+import NewRelicAPI from "new-relic-browser";
+import { useQuery } from "@apollo/react-hooks";
+import { GetUserQuery } from "gql/generated/types";
+import { GET_USER } from "gql/queries";
+import get from "lodash/get";
+import { useLocation } from "react-router-dom";
+
+declare global {
+  interface Window {
+    newrelic: typeof NewRelicAPI; // eslint-disable-line
+  }
+}
+
+type SendEvent = (
+  eventName: string,
+  attributes?: { [key: string]: string | number }
+) => void;
+
+interface Analytics {
+  sendEvent: SendEvent;
+}
+
+const AnalyticsContext = React.createContext<Analytics | null>(null);
+
+export const AnalyticsProvider: React.FC = ({ children }) => {
+  const { data } = useQuery<GetUserQuery>(GET_USER);
```

Comment:
The context provider was [moved](https://github.com/evergreen-ci/spruce/pull/206/files#diff-46ef2374da6211486f3a408db68b2bcbR53) to allow you to use `useQuery` within the analytics provider which results in having a context depend on a context. If this was a hook this also wouldn't have needed to happen.